core/tests/protractor_utils/ExplorationEditorTranslationTab.js

```diff
@@ -31,12 +31,21 @@ var ExplorationEditorTranslationTab = function() {
   this.exitTutorial = function() {
     // If the translation welcome modal shows up, exit it.
     translationWelcomeModal.isPresent().then(function(isVisible) {
-      if (isVisible) {
-        waitFor.elementToBeClickable(
-          dismissWelcomeModalButton,
-          'Welcome modal is taking too long to appear');
-        dismissWelcomeModalButton.click();
-      }
+      waitFor.visibilityOf(dismissWelcomeModalButton, 2000,
+        'Welcome modal not becoming visible').then(
```

Comment:
What is the `2000` parameter doing? `waitFor.visibilityOf` only accepts 2 parameters:

https://github.com/oppia/oppia/blob/c513a4083d7043083501d91e12e4d0f0681186ae/core/tests/protractor_utils/waitFor.js#L82

Reply:
Also you shouldn't need the `.then`. I think Protractor's control flow management should automatically handle waiting for the first `waitFor` to complete before starting the second one

Reply:
Ah, I missed this one, since usually the format for `ExpectedConditions` have three parameters. I put 2000 so that it doesn't wait for 10s to go to the error block. Removing the 2000.

Reply:
I put the `.then` to add the error function. Or are you refering to this: `translationWelcomeModal.isPresent().then(function(isVisible) {` ?

Reply:
Ahh, I see now why that's required to use the error function. Makes sense!